libs/sdk-core/src/breez_services.rs

```diff
@@ -903,7 +903,29 @@ impl BreezServices {
             let closed_channel_tx = self.closed_channel_to_transaction(closed_channel).await?;
             closed_channel_payments.push(closed_channel_tx);
         }
+        let peers = self.node_api.list_peers().await?;
+
+        for p in peers {
+            for h in p.channels {
+                let htlcs = h.htlc;
+                if let Some(hh) = htlcs {
+                    for cc in hh {
+                        info!("htlc info {:?}{:?}", cc.payment_hash, cc.expiry);
+                        let payment_hash = hex::encode(cc.payment_hash);
+
+                        if let Ok(Some(payment)) = self.persister.get_payment_by_hash(&payment_hash)
+                        {
+                            info!(
+                                "htlc payment is {:?}",
+                                payment.status == PaymentStatus::Pending
+                            );
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
This whole logic of matching the htlc to the specific pending payment should be inside the NodeAPI implementation.

Reply:
Done moved to NodeAPI